test/distributed/checkpoint/fsdp/test_fsdp_dsd.py

```diff
@@ -182,6 +187,392 @@ def _test_save_with_fsdp1_and_load_with_fsdp2(self, state_dict_type: StateDictTy
         self.assertEqual(fsdp2_full_msd, fsdp1_full_msd)
         self.assertEqual(fsdp1_full_osd, fsdp2_full_osd)
 
+    @skip_if_lt_x_gpu(4)
+    @with_temp_dir
+    def test_save_with_fsdp1_and_load_with_fsdp2_tp(self):
+        """
+        Test that we can save a model with FSDP1 and load it with FSDP2 + TP on 2d mesh.
+        """
+
+        def _get_base_model(mlp_dim: int = 2):
+            base_model = nn.Sequential(MLP(mlp_dim), MLP(mlp_dim), MLP(mlp_dim))
+            return base_model
+
+        state_dict_type_list = [
+            StateDictType.FULL_STATE_DICT,
+            StateDictType.SHARDED_STATE_DICT,
+        ]
+
+        for state_dict_type in state_dict_type_list:
```

Comment:
This seems to be redundant to me. Can we save both the FSDP1 state_dict and baseline state_dict? Basically, when calling `dcp.save()`, you can pass something like
```
state_dict = {
   "msd1": baseline_state_dict,
   "osd1": baseline_optimizer_state_dict,
   "msd2": fsdp1_state_dict,
   "osd2": fsdp1_optimizer_state_dict,
}
```
This can also simplify the test code.